Modifications to D2H custom generators for Lc

jezwilkinson opened the pull request:
Modify existing custom generator for Lc in pPb for signal training in ML studies to include option for forced decay chain of kaon from Lc (see https://github.com/alisw/AliRoot/pull/893). Also added new generator for PbPb that includes same.

chiarazampolli commented:
This was already tested, this is why I merged it.

Chiara

chiarazampolli commented:
Ciao Jeremy,

Sorry, I reverted your commit as I had not noticed that you had implemented a new "Hijing_HF001.C" file. Why did you not use the same as we changed for the test? I mean: why do you want to keep all possibility + the Lc in a separate file, and not just everything in the same? 

Chiara

jezwilkinson commented:
Ciao Chiara,
I modified the Hijing_HF001.C we used for the test and added it as "Hijing_PbPb_Lc.C" - modification was so that the same macro can also be used for the forced pKpi decay in PbPb if needed, and it seems more organised to keep the macros that are specific to the BDT training separate from the ones we use for "general" productions (since we also handled the pPb production macro in the same way).

chiarazampolli commented:
Ciao Jeremy,

But then, I would keep the Lc --> K0sp and Lc --> pKpi in the "standard" file, and have in the new (to be renamed, with a "BDT" suffix and no "Lc") only the part for the BDT MC. What do you think?

BTW, pPb had to go to a different file as the weighting for centrality (for example) is not used there.

Cheers,

Chiara

chiarazampolli commented:
Ciao Jeremy,

Sorry to bother, but we will tag most likely quite soon, so I was wondering whether you agree with my proposal for this macro. 

Thanks and cheers,

Chiara
